test/test_sklearn/test_import_sklearn.py

```diff
@@ -0,0 +1,141 @@
+"""
+Import sklearn without enough memory causes errors to be raised but these are platform
+specific. To tackle this problem, pynisher will raise a general PynisherException but
+individual errors types can be raised as a MemoryLimitException.
+"""
+import os
+import sys
+
+from pynisher import MemoryLimitException, PynisherException, limit
+from pynisher.util import Monitor
+
+import pytest
+
+# My local machine was complaining and then gave a sigsegv, raising a PynisherException
+# This prevents it
+os.environ["OPENBLAS_NUM_THREADS"] = "1"
+
+plat = sys.platform.lower()
+
+
+def import_sklearn() -> bool:
+    """Just imports sklearn"""
+    m = Monitor()
+    print("Before import ", m.memory("mb"))
+
+    import sklearn  # noqa
+
+    print("After import ", m.memory("mb"))
+
+    return True
+
+
+@pytest.mark.skipif(not plat.startswith("linux"), reason="linux specific")
+def test_import_fail_linux() -> None:
+    """
+    Expects
+    -------
+    * Should fail to import if the memory limit is too low
+
+    Note
+    ----
+    For sklearn on linux, this triggers an ImportError and so we have
```

Comment:
I don't understand this note, why is the wrap necessary?

Reply:
Supports you have two functions
```python
def f():
    import sklearn  # Will raise an ImportError if there's not enough memory to import
    
def h():
    import abaosdb
    
with limit(f, memory_limit=(50, "mb") as lf:
    lf()  # <---- raises ImportError
    

with limit(h, memory_limit=(50, "mb") as lh:
    lh()  # <---- raises ImportError
```

One of them is due to the memory limit and one is due to a user error. The options here were to treat all `ImportError` as a `MemoryLimitException` when `memory` is set, or let all `ImportError` propagate up. I went with the latter as we have no way to tell.

In SMAC this would be dealt with by `limit(f, wrap_errors=True)` in which case it just becomes a `PynisherException`, we catch it and say it crashed, move on.

At the end of the day we have no way to identify this as a real error caused by a memory limit without explicit knowledge of this behaviour with importing sklearn.

I guess this test is just a bit more explicit in trying to say that on Linux, this is how sklearn will act.